modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/DefaultExceptionEnricherFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import static org.mule.config.i18n.MessageFactory.createStaticMessage;
+import org.mule.api.MuleRuntimeException;
+import org.mule.extension.api.introspection.ExceptionEnricher;
+import org.mule.extension.api.introspection.ExceptionEnricherFactory;
+
+public class DefaultExceptionEnricherFactory implements ExceptionEnricherFactory
+{
+    private final Class<? extends ExceptionEnricher> handler;
+
+    public DefaultExceptionEnricherFactory(Class<? extends ExceptionEnricher> handlerClass)
+    {
+        handler = handlerClass;
+    }
+
+    @Override
+    public ExceptionEnricher createEnricher()
+    {
+        if (handler != null)
```

Comment:
fail fast. Use Preconditions class to throw IllegalArgumentException on the constructor if the argument is null